Add JSON console formatter

surik opened the pull request:
this commit adds `Lager.JsonFormatter` which can be used as formatter for
the console backend. Right now, it used pre-defined options and it is
not possible to change them.

Note that the formatter tries to cast nested maps and keywords.

Poison is used as JSON parser. It is easy to move to Jason when it's
needed.

In application which uses exlager console formatter can be changes via
the following option:

    log.console.format = default | json

mgumz commented:
You also upgraded Lager (+ related deps) … this mixes concerns of this one commit a little bit.
Could you please split up both the feature and the dependency upgrade?

surik committed:
Update lager to 3.6.3

surik committed:
Add JSON console formatter

This commit adds `Lager.JsonFormatter` which can be used as formatter for
the console backend. Right now, it used pre-defined options and it is
not possible to change them.

Note that the formatter tries to cast nested maps and keywords.

Poison is used as JSON parser. It is easy to move to Jason when it's
needed.

In application which uses exlager console formatter can be changes via
the following option:

    log.console.format = default | json

surik commented:
Have split into two commits.
